src/er_coap_client.erl

```diff
@@ -112,7 +120,7 @@ split_segments(Path, Char, Acc) ->
     end.
 
 make_segment(Seg) ->
-    list_to_binary(http_uri:decode(Seg)).
+    list_to_binary(uri_string:percent_decode(Seg)).
```

Comment:
If I remember correctly, this is not compatible.

Reply:
http://erlang.2086793.n4.nabble.com/http-uri-decode-equivalent-in-uri-string-module-td4727306.html